packages/cli/src/services/snapshot-service.ts

```diff
@@ -0,0 +1,65 @@
+import * as fsAsync from 'node:fs/promises'
+import * as fs from 'node:fs'
+import * as path from 'node:path'
+import * as stream from 'node:stream/promises'
+
+import { checklyStorage } from '../rest/api'
+import { findFilesRecursively, pathToPosix } from './util'
+
+export interface Snapshot {
+  key: string,
+  path: string,
+}
+
+export async function pullSnapshots (basePath: string, snapshots?: Snapshot[] | null) {
+  if (!snapshots?.length) {
+    return
+  }
+
+  try {
+    for (const snapshot of snapshots) {
+      const fullPath = path.resolve(basePath, snapshot.path)
+      if (!fullPath.startsWith(basePath)) {
+        // The snapshot file should always be within the project, but we validate this just in case.
+        throw new Error(`Detected invalid snapshot file ${fullPath}`)
+      }
+      await fsAsync.mkdir(path.dirname(fullPath), { recursive: true })
+      const fileStream = fs.createWriteStream(fullPath)
+      const { data: contentStream } = await checklyStorage.download(snapshot.key)
+      contentStream.pipe(fileStream)
+      await stream.finished(contentStream)
+    }
+  } catch (err: any) {
+    throw new Error(`Error downloading snapshots: ${err.message}`)
+  }
+}
+
+export function detectSnapshots (projectBasePath: string, scriptFilePath: string) {
```

Comment:
TODO:

Playwright is able to maintain different snapshot files for different platforms (f.ex `*-chromium-linux.png` for Chromium Linux, which is what we use in our runtimes).  

This current implementation will detect all of the snapshots, though, rather than just pushing the Chromium Linux ones.